### DISCUSSION TITLE
4. 네트워크 - 데이터 계층

### DISCUSSION THREAD

**Discussion by seongeun42** [Shortudy-10th/tech-for-developer#158] [질의 응답]
데이터 계층 질의응답

**Comment by seongeun42** [comment]
## Q. 데이터 링크 계층의 역할은 무엇인가요?

데이터 링크 계층은 물리 계층을 이용해 네트워크 상의 인접한 장치 간의 데이터를 신뢰성 있게 전송하는 역할을 합니다.

**Comment by seongeun42** [comment]
## Q. MAC 주소란 무엇인가요?

MAC 주소란 컴퓨터나 라우터 등의 통신 장치에 할당된 고유 식별자로, 데이터 링크 계층에서 통신을 위해 사용되는 물리적 주소입입니다.

**Comment by seongeun42** [comment]
## Q. ARP에 대해 설명해주세요.

ARP는 네트워크 상에서 IP주소를 MAC주소로 대응시키기 위해 사용하는 프로토콜입니다. 반대로, MAC주소를 IP주소로 대응시키기 위해 사용하는 프로토콜은 RARP가 있습니다.

**Comment by zini9188** [comment]
## Q. 순환 중복 검사에 대해 설명해주세요

데이터를 전송할 때 전송한 데이터에 오류가 있는지 확인하는 방법입니다. 데이터를 전송할 때 데이터 값에 따른 crc값을 계산하여 함께 전송하고, 전송 후에 받은 crc값과 비교하여 전송 과정이 잘 이뤄졌는지 확인할 수 있습니다.

** CRC 계산 방법
polynomial : x^8 + x^2 +  x + 1와 같이 지수 형태의 식

1. 데이터를 비트로 변환
2. polynomial을 비트로 변환
3. bitwise xor 계산

**Comment by zini9188** [comment]
## Q. 프레임에 대해 설명하고 패킷과 다른점을 말해주세요.

프레임이란 데이터 링크 계층에서 사용되는 단위로, 패킷을 캡슐화하며 MAC 주소를 헤더로 가지는 블록입니다.

패킷은 전송 계층에서 사용되는 단위로 세그먼트를 캡슐화하며 IP 주소를 헤더로 가지는 블록입니다.


** 프레임 구조 
프리엠블 (Preemble) : 1과 0으로 이루어진 동기 신호, 7 Byte
프레임 시작 신호 (SOF) : 프레임의 시작을 알리는데 사용, 1 Byte
목적지 MAC 주소 : 목적지의 맥주소를 나타냄, 6 Byte
출발지 MAC 주소 : 출발지의 맥주소를 나타냄, 6 Byte
유형/길이 : 프레임의 데이터 필드 길이나 프로토콜의 종류 표시, 2 Byte
데이터 : 실제 전송되는 데이터로 46바이트 이하인 경우 패딩하여 전송, 46-1500 Byte
FCS : 에러 발생 여부를 확인하기 위한 필드, 4 Byte

** 프레임의 크기는 SOF와 프리엠블을 제외하여 계산한다.

**Comment by watty-exclamationMark** [comment]
## 이더넷에 대해 말해주세요.
이더넷은 데이터 링크 계층에서 사용되는 프로토콜 중 하나로, 대부분 LAN 환경에서 표준으로 사용됩니다.
이더넷 헤더(목적지 MAC Addr, 출발지 MAC, Addr 유형)과 트레일러가 추가된 데이터를 프레임이라고 합니다.

**Comment by watty-exclamationMark** [comment]
## 스위치와 브리지
하나의 네트워크망 안의 서로 다른 LAN을 연결하여 LAN을 확장 구성하는 장비입니다.
